pkgs/development/libraries/mesa/default.nix

```diff
@@ -90,124 +89,126 @@ let self = stdenv.mkDerivation {
   #  revive ./dricore-gallium.patch when it gets ported (from Ubuntu), as it saved
   #  ~35 MB in $drivers; watch https://launchpad.net/ubuntu/+source/mesa/+changelog
   patches = [
-    ./symlink-drivers.patch
     ./missing-includes.patch # dev_t needs sys/stat.h, time_t needs time.h, etc.-- fixes build w/musl
-    ./disk_cache-include-dri-driver-path-in-cache-key.patch
   ] ++ lib.optional stdenv.isDarwin ./darwin-clock-gettime.patch;
 
   outputs = [ "out" "dev" "drivers" ]
             ++ lib.optional (elem "swrast" galliumDrivers) "osmesa";
 
   # TODO: Figure out how to enable opencl without having a runtime dependency on clang
-  configureFlags = [
+  mesonFlags = [
+    # https://gitlab.freedesktop.org/search?utf8=✓&snippets=&scope=&search=get_option('datadir')&project_id=176
+    "--datadir=${placeholder "out"}/share" # "vendorjsondir=\${out}/share/glvnd/egl_vendor.d" https://gitlab.freedesktop.org/mesa/mesa/blob/master/src/egl/meson.build#L156
+    # NOTE: Useless since -Dgallium-opencl=disabled?
+    # https://gitlab.freedesktop.org/search?utf8=✓&snippets=&scope=&search=get_option('sysconfdir')&project_id=176
```

Comment:
This will break xmlutil:

https://gitlab.freedesktop.org/mesa/mesa/blob/f3247841040a202faffe4709c07da9bd41693580/src/util/meson.build#L135

We need to use 

```nix
    "--sysconfdir=/etc"
```

And fix the gallium thing in a patch.

Reply:
`-Dgallium-opencl`?

Reply:
We would need to patch [this line](https://gitlab.freedesktop.org/mesa/mesa/blob/3090c6b9e9c114b3438be83399c7bed12f920505/src/gallium/targets/opencl/meson.build#L71), replacing `get_option('sysconfdir')` with `get_option('prefix'), 'etc'`

Reply:
So `-Dgallium-opencl=icd` instead of `-Dgallium-opencl=disabled` now? Since it seems like that build only processes that part if we compile with `icd` [1].

[1] https://gitlab.freedesktop.org/mesa/mesa/blob/3090c6b9e9c114b3438be83399c7bed12f920505/meson.build#L666

Reply:
Sorry, I am only familiar with meson, not mesa. I have no idea if this is actually desirable – only that if it is, we need to patch it.

Reply:
Probably in the future so I'll include it.